app/ante.go

```diff
@@ -59,16 +61,17 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 		ante.NewTxTimeoutHeightDecorator(),
 		ante.NewValidateMemoDecorator(options.AccountKeeper),
 		ante.NewConsumeGasForTxSizeDecorator(options.AccountKeeper),
-		// custom archway fee deduction, which splits fees between gastracker and auths fee collector
-		gastrackerante.NewDeductFeeDecorator(options.AccountKeeper, options.BankKeeper, options.FeegrantKeeper, options.GasTrackingKeeper),
 		// SetPubKeyDecorator must be called before all signature verification decorators
 		ante.NewSetPubKeyDecorator(options.AccountKeeper),
 		ante.NewValidateSigCountDecorator(options.AccountKeeper),
 		ante.NewSigGasConsumeDecorator(options.AccountKeeper, sigGasConsumer),
 		ante.NewSigVerificationDecorator(options.AccountKeeper, options.SignModeHandler),
 		ante.NewIncrementSequenceDecorator(options.AccountKeeper),
 		ibcante.NewAnteDecorator(options.IBCChannelkeeper),
-		gastrackerante.NewTxGasTrackingDecorator(options.GasTrackingKeeper),
+		// Custom Archway interceptor to track new transactions
+		trackingAnte.NewTxGasTrackingDecorator(options.TrackingKeeper),
+		// Custom Archway fee deduction, which splits fees between x/rewards and x/auth fee collector
+		rewardsAnte.NewDeductFeeDecorator(options.AccountKeeper, options.BankKeeper, options.FeegrantKeeper, options.RewardsKeeper),
```

Comment:
I would put the trackingAnte and rewardsAnte before `SetPubKeyDecorator` to keep the same SDK weighted flow for antes.

Reply:
Done